source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRFlowLayout.mm

```diff
@@ -0,0 +1,313 @@
+}
+
+- (instancetype)initWithStyle:(ACRContainerStyle)style
+                  parentStyle:(ACRContainerStyle)parentStyle
+                   hostConfig:(ACOHostConfig *)acoConfig
+                    superview:(UIView *)superview
+{
+    std::shared_ptr<HostConfig> config = [acoConfig getHostConfig];
+    self = [self initWithFrame:superview.frame attributes:nil];
+    if (self) {
+        _style = style;
+        if (style != ACRNone &&
+            style != parentStyle) {
+            self.backgroundColor = [acoConfig getBackgroundColorForContainerStyle:_style];
+            [self setBorderColorWithHostConfig:config];
+        }
+    }
+    return self;
+}
+
+- (instancetype)initWithFrame:(CGRect)frame attributes:(nullable NSDictionary<NSString *, id> *)attributes
+{
+     return [super initWithFrame:CGRectMake(0, 0, frame.size.width, frame.size.height)];
+}
+
+- (void)setUpGrid
+{
+    _verticalStack = [self createVerticalStack];
+    [self addNewRowWithView:nil];
+}
+
+- (void)addNewRowWithView:(UIView *)view
+{
+    _horizontalStack = [self createHorizontalStack];
+    [_verticalStack addArrangedSubview:_horizontalStack];
+    if(view)
+    {
+        [_horizontalStack addArrangedSubview:view];
+    }
+}
+
+- (void)addViewInCurrentRow:(UIView *)view
+{
+    [_horizontalStack addArrangedSubview:view];
+}
+
+- (UIStackView *)createHorizontalStack
+{
+    UIStackView *stackView = [[UIStackView alloc] init];
+    stackView.axis = UILayoutConstraintAxisHorizontal;
+    stackView.alignment = UIStackViewAlignmentFill;
+    stackView.distribution = UIStackViewDistributionFill;
+    stackView.spacing = _columnSpacing;
+    stackView.translatesAutoresizingMaskIntoConstraints = NO;
+    return stackView;
+}
+
+-(UIStackViewAlignment) horizontalAlignment
+{
+    if(_layout->GetItemFit() == ItemFit::Fill)
+    {
+        return UIStackViewAlignmentFill;
+    }
+    
+    switch (_layout->GetHorizontalAlignment()) {
+        case AdaptiveCards::HorizontalAlignment::Center:
+            return UIStackViewAlignmentCenter;
+            
+        case AdaptiveCards::HorizontalAlignment::Right:
+            return UIStackViewAlignmentTrailing;
+            
+        case AdaptiveCards::HorizontalAlignment::Left:
+            return UIStackViewAlignmentLeading;
+    }
+    
+}
+
+- (UIStackView *)createVerticalStack
+{
+    UIStackView *stackView = [[UIStackView alloc] init];
+    stackView.axis = UILayoutConstraintAxisVertical;
+    stackView.alignment = [self horizontalAlignment];
+    stackView.distribution = UIStackViewDistributionEqualSpacing;
+    stackView.spacing = _rowSpacing;
+    
+    [self addSubview:stackView];
+    
+    stackView.translatesAutoresizingMaskIntoConstraints = NO;
+    
+    NSLayoutConstraint *selfWidth = [self.widthAnchor constraintEqualToConstant:_availableRowSpace];
+    selfWidth.priority = 999;
+    NSLayoutConstraint *leading = [stackView.leadingAnchor constraintEqualToAnchor:self.leadingAnchor constant:0];
```

Comment:
since constant is set to 0 here we can use constraintEqualToAnchor and not specify the constant here.

Reply:
same below.